include/aws/http/connection.h

```diff
@@ -452,6 +462,14 @@ enum aws_http_version aws_http_connection_get_version(const struct aws_http_conn
 AWS_HTTP_API
 struct aws_channel *aws_http_connection_get_channel(struct aws_http_connection *connection);
 
+/**
+ * Helper function that returns an empty map can be used in aws_http_client_connection_options as alpn_string_map.
+ * The caller owns the returned map.
+ * The aws_string created as the key of the map will be owned by the map.
+ */
+AWS_HTTP_API
+int aws_http_alpn_map_init(struct aws_allocator *allocator, struct aws_hash_table *map);
```

Comment:
Not sure what you were trying to say with "The aws_string created as the key of the map will be owned by the map." ??

how about just:
```
/**
 * Initialize an empty hash-table that maps `struct aws_string *` to `enum aws_http_version`.
 * This map can used in aws_http_client_connections_options.alpn_string_map.
 */
```